pkg/secret/factory.go

```diff
@@ -195,18 +195,22 @@ func (f *Factory) makeRegistrySecret(secretName string, namespace string) (*core
 		return nil, "", err
 	}
 
-	reg := f.Registry
+	registry := f.Registry
 	// Handle path in registry
-	if strings.ContainsRune(reg, '/') {
-		r, err := name.NewRepository(reg, name.WeakValidation)
-		if err != nil {
-			return nil, "", err
+	if strings.ContainsRune(registry, '/') {
+		if strings.Contains(registry, "https://index.docker.io") {
```

Comment:
Let's loosen this to just `index.docker.io`, given that we refer to it as `ghcr.io` instead of `https://ghcr.io`.